pkg/scheduler/reconciliation/reconciliation_test.go

```diff
@@ -729,3 +729,93 @@ func dbConnection(t *testing.T) db.Connection {
 	}
 	return dbConn
 }
+
+func TestReconciliationParallel(t *testing.T) {
+
+	t.Run("Create multiple instances of single reconciliations in parallel", func(t *testing.T) {
+		repo := newPersistentRepository(t)
+
+		inventory, err := cluster.NewInventory(dbConnection(t), true, cluster.MetricsCollectorMock{})
+		require.NoError(t, err)
+
+		errChannel := make(chan error, 100)
+		mockClusterState, _ := createClusterStates(t, inventory)
+
+		defer func() {
+			require.NoError(t, inventory.Delete(mockClusterState.Cluster.RuntimeID))
+		}()
+
+		startAt := time.Now().Add(1 * time.Second)
+		for i := 0; i < 50; i++ {
+			go func() {
+				time.Sleep(time.Until(startAt))
+				_, err := repo.CreateReconciliation(mockClusterState, nil)
+				if err != nil {
+					errChannel <- err
+				}
+			}()
+		}
+		time.Sleep(5 * time.Second)
```

Comment:
We can also use wait group here to eliminate the uncertainty of running time (the time mainly in spinning up database for you test, it depends on many hardware conditions).

Reply:
Done